### PR TITLE
internal: tweak moon run single .mbt file

### PR DESCRIPTION
**Pull request by Young-Flash**

### CODE REVIEW

File: crates/moon/src/cli/run.rs [hunk 1]
```diff
@@ -202,7 +206,18 @@ pub fn run_run_internal(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Res
 
     let sort_input = cmd.build_flags.sort_input;
 
-    let package_path = cmd.package_or_mbt_file.clone();
+    // run .mbt inside a package should run as a package
+    let package_path = if cmd.package_or_mbt_file.ends_with(".mbt") {
+        // `package_path` based on `source_dir`
+        dunce::canonicalize(std::env::current_dir()?.join(cmd.package_or_mbt_file))?
```
Comment (lijunchen) [inline crates/moon/src/cli/run.rs]: canonicalize a nonexistent path will cause an error, add context for better error reporting

File: crates/moon/src/cli/run.rs [hunk 2]
```diff
@@ -90,15 +91,14 @@ fn run_single_mbt_file(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Resu
         .map_or(TargetBackend::default(), |it| *it);
     let core_bundle_path = moonutil::moon_dir::core_bundle(target_backend);
 
-    // `parent_path` is not always same with `current_dir`, since `cmd.package_or_mbt_file` can be something like "a/b/c/single.mbt"
-    let parent_path = mbt_file_path.parent().unwrap();
+    let output_artifact_path = std::env::temp_dir().join("moon_run_artifact");
```
Comment (lijunchen) [inline crates/moon/src/cli/run.rs]: use `tempfile` crate?

File: crates/moon/src/cli/run.rs [hunk 3]
```diff
@@ -76,7 +77,7 @@ pub fn run_run(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Result<i32>
 }
 
 fn run_single_mbt_file(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Result<i32> {
-    let current_dir = std::env::current_dir().unwrap();
+    let current_dir = std::env::current_dir()?;
     let mbt_file_path = current_dir.join(cmd.package_or_mbt_file);
 
     if !mbt_file_path.exists() || !mbt_file_path.is_file() {
```
Comment (lijunchen) [inline crates/moon/src/cli/run.rs]: `if !mbt_file_path.exists() || !mbt_file_path.is_file()`

=> 

`if !mbt_file_path.is_file()`

File: crates/moon/src/cli/run.rs [hunk 4]
```diff
@@ -117,7 +117,7 @@ fn run_single_mbt_file(cli: &UniversalFlags, cmd: RunSubcommand) -> anyhow::Resu
     let link_core_command = [
         "link-core",
         &core_bundle_path.join("core.core").display().to_string(),
```
Comment (lijunchen) [inline crates/moon/src/cli/run.rs]: `core_bundle_path.join("core.core")` => `core_core()`

File: crates/moon/src/cli/run.rs (hunk 2 above)
Comment (lijunchen) [reply crates/moon/src/cli/run.rs]: and no need to create `moon_run_artifact` if use `tempfile`

Comment (Young-Flash) [reply crates/moon/src/cli/run.rs]: we want the artifact to be placed in a known path, ide debug need it

Comment (lijunchen) [reply crates/moon/src/cli/run.rs]: If so, I suggest placing temp files in `.moon`.

Comment (lijunchen) [reply crates/moon/src/cli/run.rs]: It seems that `std::env::temp_dir()` returns the same directory each time. What will happen if there is a name conflict in `moon_run_artifact`?

**Pull request merged**